x-pack/plugins/maps/public/kibana_services.js

```diff
@@ -141,15 +135,30 @@ let dataSearchService;
 export const setSearchService = searchService => (dataSearchService = searchService);
 export const getSearchService = () => dataSearchService;
 
+let kibanaVersion;
+export const setKibanaVersion = version => (kibanaVersion = version);
+export const getKibanaVersion = () => kibanaVersion;
+
 let mapConfig;
 export const setMapConfig = config => (mapConfig = config);
 export const getMapConfig = () => mapConfig;
 
-export const getIsEmsEnabled = () => getMapConfig().includeElasticMapsService;
-export const getEmsFontLibraryUrl = () => getMapConfig().emsFontLibraryUrl;
-export const getEmsTileLayerId = () => getMapConfig().emsTileLayerId;
-export const getEmsFileApiUrl = () => getMapConfig().emsFileApiUrl;
-export const getEmsTileApiUrl = () => getMapConfig().emsTileApiUrl;
-export const getEmsLandingPageUrl = () => getMapConfig().emsLandingPageUrl;
-export const getRegionmapLayers = () => _.get(getMapConfig(), 'regionmap.layers', []);
-export const getTilemap = () => _.get(getMapConfig(), 'tilemap', []);
+export const getEnabled = () => getMapConfig().enabled;
+export const getShowMapVisualizationTypes = () => getMapConfig().showMapVisualizationTypes;
+export const getShowMapsInspectorAdapter = () => getMapConfig().showMapsInspectorAdapter;
+export const getPreserveDrawingBuffer = () => getMapConfig().preserveDrawingBuffer;
+export const getEnableVectorTiles = () => getMapConfig().enableVectorTiles;
+export const getProxyElasticMapsServiceInMaps = () => getMapConfig().proxyElasticMapsServiceInMaps;
+
+let legacyMapConfig;
```

Comment:
What is the difference between legacyMapConfig and mapConfig? Maybe these could be unified in the kibana services? Or at least use better names.

Reply:
This relates to https://github.com/elastic/kibana/issues/64810. In NP, config values are served up by plugins as defined in `kibana.json` > `configPath` with no two plugins being allowed the same config path. For better explanation than I can provide here, see [this](https://github.com/elastic/kibana/blob/master/src/core/MIGRATION.md#configure-plugin).

In the Maps app, we're heavily leveraging config values defined in both `map.*` which is served up by `maps_legacy` and `xpack.maps.*` which is served up by the `maps` app. They come from two different places, `map.*` config values are provided   through `maps_legacy` as a plugin dependency in Maps. Not ideal, but really no different than how it's been up to this point, the relationship is just more explicit in NP. `xpack.maps.*` config values are "given" to the plugin through the `initializerContext` passed through the constructor on instantiation.

I considered consolidating them under one `config`, but ultimately decided not to since the rest of the app doesn't "see" where they come from but rather just uses the correct getter. And given the differentness in where we get these config values, it made sense to me to keep the initial service setting separate for each. Having said that, I'm pretty open to renaming if it helps or even consolidating them if you feel strongly. If you want to rename, any suggestions?

Reply:
How about `mapsLegacyConfig` with a comment "kibana.yml settings from maps_legacy plugin, map.*"

And `mapsConfig` with a comment "kibana.yml settings from this plugin, xpack.maps.*"

Reply:
Updated suggestions to 

`kibanaCommonMapConfig` with a comment "map.* kibana.yml settings from maps_legacy plugin that are shared between OSS map visualizations and maps app."

And `mapAppConfig` with a comment "xpack.maps.* kibana.yml settings from this plugin"